src/game/core/entity/Entity.ts

```diff
@@ -1,31 +1,65 @@
-import { TCoordinates, TSize } from 'game/core/types';
+import { TCoordinates, TSize, TSpeedModifierRefernce } from 'game/core/types';
 import { Stage, Sprite } from 'game/core';
 
 export default class Entity {
-  private _sprite: Sprite;
+  sprite: Sprite;
   position: TCoordinates;
-  speed: number | null
+  speed: number | null;
+  speedModifierRefernce?: TSpeedModifierRefernce;
+  hitBoxStartPoint?: TCoordinates | null;
+  hitBoxSize?: TSize | null;
 
-  constructor(initialPosition: TCoordinates, sprite: Sprite, speed?: number) {
-    this._sprite = sprite;
+  constructor(props: {
+    initialPosition: TCoordinates,
+    sprite: Sprite,
+    speed?: number,
+    speedModifierRefernce?: TSpeedModifierRefernce,
+    hitBoxStartPoint?: TCoordinates,
+    hitBoxSize?: TSize,
+  }
+  ) {
```

Comment:
❗️
Давай вынесем типизацию пропсов в отдельный интерфейс в рамках компонента, а то сейчас декларация функции раздута
```suggestion
  interface EntityProps {
    initialPosition: TCoordinates;
    sprite: Sprite;
    speed?: number;
    speedModifierRefernce?: TSpeedModifierRefernce;
    hitBoxStartPoint?: TCoordinates;
    hitBoxSize?: TSize;
  }

  constructor(props: EntityProps) {
```

Reply:
fixed